src/omotes_sdk/omotes_interface.py

```diff
@@ -184,14 +193,68 @@ def connect_to_submitted_job(
         :param auto_disconnect_on_result: Remove/disconnect from all queues pertaining to this job
             once the result is received and handled without exceptions through
             `callback_on_finished`.
+        :param auto_dead_letter_after_ttl: When erroneous situations occur (e.g. client is offline),
+            the job result message (if available) will be dead lettered after the given TTL,
+            and all queues of this job will be removed subsequently. Default to 48 hours if unset.
+            Set to `None` to turn off auto dead letter and clean up, but be aware this may lead to
+            messages and queues to be stored in RabbitMQ indefinitely
+            (which uses up memory & disk space).
+        :param reconnect: When True, first check the job queues status and raise an error if not
+            exist. Default to True.
         """
+        job_results_queue_name = OmotesQueueNames.job_results_queue_name(job.id)
+        job_progress_queue_name = OmotesQueueNames.job_progress_queue_name(job.id)
+        job_status_queue_name = OmotesQueueNames.job_status_queue_name(job.id)
+
+        if reconnect:
+            logger.info("Reconnect to the submitted job %s is set to True. "
+                        + "Checking job queues status...", job.id)
+            if not self.broker_if.queue_exists(job_results_queue_name):
+                raise RuntimeError(
+                    f"The {job_results_queue_name} queue does not exist or is removed. "
+                    "Abort reconnecting to the queue."
+                )
+            if (callback_on_progress_update
+                    and not self.broker_if.queue_exists(job_progress_queue_name)):
+                raise RuntimeError(
+                    f"The {job_progress_queue_name} queue does not exist or is removed. "
+                    "Abort reconnecting to the queue."
+                )
+            if (callback_on_status_update
+                    and not self.broker_if.queue_exists(job_status_queue_name)):
+                raise RuntimeError(
+                    f"The {job_status_queue_name} queue does not exist or is removed. "
+                    "Abort reconnecting to the queue."
+                )
+
         if auto_disconnect_on_result:
             logger.info("Connecting to update for job %s with auto disconnect on result", job.id)
             auto_disconnect_handler = self._autodelete_progres_status_queues_on_result
         else:
             logger.info("Connecting to update for job %s and expect manual disconnect", job.id)
             auto_disconnect_handler = None
 
+        # TODO: handle reconnection after the message is dead lettered but queue still exists.
```

Comment:
This bothered me and needs some discussion.

**Issue**: If the SDK reconnects too late (see scenario 4 below), e.g. after the time delta of `worker processing time + message TTL`, the job result message is already dead lettered, but queues still exist (so no reconnection issue). The SDK could be waiting for an empty job result until it is timeout. However, the job progress message will likely be consumed by SDK once reconnected and show the message such as `Job e8de982a-d46d-405a-87be-d1056936c964 progress (type: grow_optimizer_default). Progress: 1.0, message: Calculation finished.`

![TTL scenarios](https://github.com/user-attachments/assets/441fbb66-aae1-41fb-ba7f-dd873ce0655e)

Reply:
Nice diagrams! Could you share the tool you used for this? :O

Diving into scenario 4, I believe the timeout will not be triggered. In our situation by the time the TTL activates any timeout should have already been triggered by the orchestrator and any job results send. However, I hope the queue TTL will trigger so the queue is removed and the SDK is notified. If the queue TTL is reset, this may lead in an indefinite wait.

Good catch!

Reply:
Btw, lets split it to a separate ticket so we can merge this PR soonish.

Reply:
Draw.io :) 

Good idea, let's avoid PR from growing and opening forever... I created an issue here https://github.com/Project-OMOTES/omotes-sdk-python/issues/67

Reply:
Hehehe it seems your draw.io skills are superior to mine! 😄